apollo-router/src/query_planner/fetch.rs

```diff
@@ -61,6 +61,26 @@ impl OperationKind {
     }
 }
 
+impl From<OperationKind> for apollo_compiler::ast::OperationType {
```

Comment:
`OperationKind` is the internal type on the router side, that comes from the query plan IIRC

Reply:
it comes from the query planner's `FetchNode` structure: https://github.com/apollographql/router/blob/dev/apollo-router/src/query_planner/fetch.rs#L67-L86
For now we are tightly integrated with it, but if apollo-compiler's type can deserialize in the same way then we can use it instead

Reply:
there's duplication here mainly because that `OperationKind` predates the apollo-compiler by a very long time, and we have not updated everything in the process

Reply:
ahh cool cool one more thing to look forward to simplifying with this replatforming